sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/ChangeFeedFetcher.java

```diff
@@ -115,18 +118,22 @@ private Mono<FeedResponse<T>> nextPageInternal(DocumentClientRetryPolicy retryPo
                        FeedRangeContinuation continuationSnapshot =
                            this.changeFeedState.getContinuation();
 
-                       if (this.completeAfterAllCurrentChangesRetrieved) {
+                       if (this.completeAfterAllCurrentChangesRetrieved || this.endLSN != null) {
                            if (continuationSnapshot != null) {
-                               //track the end-LSN available now for each sub-feedRange and then find the next sub-feedRange to fetch more changes
-                               boolean shouldComplete = continuationSnapshot.hasFetchedAllChangesAvailableNow(r);
+
+                               //track the end-LSN for each sub-feedRange and then find the next sub-feedRange to fetch more changes
+                               boolean shouldComplete = continuationSnapshot.hasFetchedAllChanges(r, endLSN);
                                if (shouldComplete) {
                                    this.disableShouldFetchMore();
                                    return Mono.just(r);
                                }
 
-                               if (ModelBridgeInternal.<T>noChanges(r)) {
+                               // We will return if we have seen either a 304 or we
```

Comment:
I think this is eebatable - IMO if endLsn is present we should not stop on 304 ß could just indicate lagging replica

Reply:
Deifnitely in Spark  but I would argue even in core SDK the presence of endLSN should mean we do not stop processing on 304

Reply:
Changed it so that in the core sdk we only stop when reaching the endlsn